cinema/views.py

```diff
@@ -1,45 +1,105 @@
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
+class GenreDetail(views.APIView):
+    def get(self, request, pk):
+        genre = generics.get_object_or_404(Genre, id=pk)
+        serializer = GenreSerializer(genre)
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-@api_view(["GET", "PUT", "DELETE"])
-def movie_detail(request, pk):
-    movie = get_object_or_404(Movie, pk=pk)
+    def put(self, request, pk):
+        genre = generics.get_object_or_404(Genre, id=pk)
+        serializer = GenreSerializer(genre, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "GET":
-        serializer = MovieSerializer(movie)
+    def patch(self, request, pk):
+        genre = generics.get_object_or_404(Genre, id=pk)
+        serializer = GenreSerializer(genre, data=request.data, partial=True)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
         return Response(serializer.data, status=status.HTTP_200_OK)
 
-    if request.method == "PUT":
-        serializer = MovieSerializer(movie, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data, status=status.HTTP_200_OK)
+    def delete(self, request, pk):
+        genre = generics.get_object_or_404(Genre, id=pk)
+        genre.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
 
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+class ActorList(generics.GenericAPIView,
+                mixins.ListModelMixin,
+                mixins.CreateModelMixin):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
 
-    if request.method == "DELETE":
-        movie.delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+    def get(self, request):
+        return self.list(request)
+
+    def post(self, request):
+        return self.create(request)
+
+
+class ActorDetail(generics.GenericAPIView,
+                  mixins.RetrieveModelMixin,
+                  mixins.DestroyModelMixin,
+                  mixins.UpdateModelMixin):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+    def get(self, request, pk):
+        return self.retrieve(request, pk)
+
+    def delete(self, request, pk):
+        return self.destroy(request, pk)
+
+    def put(self, request, pk):
+        return self.update(request, pk)
+
+    def patch(self, request, pk):
+        return self.partial_update(request, pk)
+
+
+class CinemaHallViewSet(viewsets.GenericViewSet,
+                        mixins.ListModelMixin,
+                        mixins.DestroyModelMixin,
+                        mixins.UpdateModelMixin,
+                        mixins.CreateModelMixin,
+                        mixins.RetrieveModelMixin):
```

Comment:
For classes that inherit from multiple classes, PEP 8 recommends placing each base class on a separate line for better readability
 
 class MyViewSet(
    viewsets.GenericViewSet,
    mixins.ListModelMixin,
    mixins.DestroyModelMixin,
    mixins.UpdateModelMixin,
    mixins.CreateModelMixin,
    mixins.RetrieveModelMixin
):